src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMultiFileCompilationModuleGroup.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using ILCompiler.DependencyAnalysis;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler
+{
+    public abstract class ReadyToRunMultiFileCompilationModuleGroup : CompilationModuleGroup
```

Comment:
We were previously using `MultiFileCompilationModuleGroup` because it gave us the right semantics of not compiling everything in all referenced assemblies (Single file mode in CoreRT means compile everything in the input assembly plus everything in all assemblies we reference into one output binary). "multifile" is a CoreRT concept that we should not bring over to ready-to-run. In ready-to-run, we'll have this mode you're adding support for, which I suggest naming something like `ReadyToRunSingleAssemblyCompilationModuleGroup`. In the future, we'll want to support multiple assemblies in a single version bubble and may be able to use `CompilationModuleGroup`s for that.